crates/sui-json-rpc/src/transaction_builder_api.rs

```diff
@@ -77,6 +79,106 @@ impl DataReader for AuthorityStateDataReader {
     }
 }
 
+/// Gas Station Client abstraction
```

Comment:
Honestly unsure where this implementation should go, I can move it into a different file or leave it here?